systeminit/si
# feat(dal): add prefix validation

## Commit Description

Validations:
- Add prefix validation mode for "validateString"
- Rename "validateStringEquals" to "validateString"
- Add ability to hide the expected strings list for "validateStringInStringArray" if an error is returned

Builtins:
- Add ImageId validation for both EC2 and AMI (use the new prefix mode)
- Add Instance validation for EC2 (use the ability to hide expected strings list if an error is returned)

Misc:
- Add ctx ownership recommendation to DEVELOPING when commiting txns within tests
- Rename schema builtins test harness props collection methods and remove doc comments as a result (now, more clear)

## GIF and Linear

<img src="https://media3.giphy.com/media/3orieV94MAL3OLlOAU/giphy.gif"/>

Fixes ENG-581
Fixes ENG-583
